### ISSUE TITLE
instants() needs to support ranges of valid values

### ISSUE THREAD

**Issue by rpolton** [sageserpent-open/americium#45]
Similarly to integers(), longs() etc, instants() should provide an overloaded factory that accepts a lower bound and an upper bound. Instant MAX defines the maximum supported Instant, '1000000000-12-31T23:59:59.999999999Z' and this is far in the future. The issue is that the standard DateTimeFormatter renders any year larger than 9999 with a '+' prefix which is potentially unexpected. In order to avoid having to code for the year 10000 (I'd like to think that Java will have been updated to handle 5-digit years consistently by the time we approach that point) it would be better to restrict the range of possible values in the factory method. The application of a filter() after construction works sometimes but often simply produces an empty stream -> Precondition violation in Trials.

**Comment by sageserpent-open** [comment]
There is a `TrialsApi.stream` method that takes a `CaseFactory` that allows this to be done (mapping to `Instant` also works off `Trials.longs`) - but it would be nice to have parity between `Trials.instants` and `Trials.integers` / `Trials.characters`.

**Comment by sageserpent-open** [comment]
Released in version 1.4.6, GIT SHA: 13474b963286baa851d79771abd17a897fb40d13

**Comment by sageserpent-open** [comment]
Evidence made using JShell running against Americium version 1.4.6 follows...

Code:

```java
import com.sageserpent.americium.java.Trials;
import com.sageserpent.americium.java.TrialsApi;

import java.time.Instant;

import static org.junit.Assert.assertThrows;

final TrialsApi api = Trials.api();

final Instant shrinkageTarget = Instant.now();

System.out.println("Shrinkage target inside the interval:");

assertThrows(RuntimeException.class, () -> api.instants(shrinkageTarget.minusSeconds(60 * 100), shrinkageTarget.plusSeconds(60 * 1000), shrinkageTarget).withLimit(10).supplyTo(caze -> {
    System.out.println(caze);
    throw new RuntimeException();
}));

System.out.println("Shrinkage target at high end of interval:");

assertThrows(RuntimeException.class, () -> api.instants(shrinkageTarget.minusSeconds(60 * 100), shrinkageTarget, shrinkageTarget).withLimit(10).supplyTo(caze -> {
    System.out.println(caze);
    throw new RuntimeException();
}));

System.out.println("Shrinkage target at low end of interval:");

assertThrows(RuntimeException.class, () -> api.instants(shrinkageTarget, shrinkageTarget.plusSeconds(60 * 1000), shrinkageTarget).withLimit(10).supplyTo(caze -> {
    System.out.println(caze);
    throw new RuntimeException();
}));

System.out.println("Shrinkage target is implicitly defined at the epoch.");

assertThrows(RuntimeException.class, () -> api.instants(Instant.EPOCH.minusSeconds(2), Instant.EPOCH.plusSeconds(1)).withLimit(10).supplyTo(caze -> {
    System.out.println(caze);
    throw new RuntimeException();
}));
```

Output:
